src/main/java/com/example/benefit/service/BenefitService.java

```diff
@@ -35,12 +39,17 @@ public BenefitDTO createAndUpdateBenefit(BenefitDTO benefitDTO) {
             benefit.setCalculationMethod(calculationMethodRepository.findByName(calculationMethodName));
         }
 
-        if (benefitRepository.findByName(benefit.getName()) == null) {
-            benefitRepository.save(benefit);
-            return benefitMapper.entityToDto(benefit);
-        } else {
-            return benefitMapper.entityToDto(benefitRepository.findByName(benefit.getName()));
-        }
+        benefitRepository.save(benefit);
```

Comment:
Let's assign saved entity to some variable and then we can create DTO from saved entity and build dto for mq from saved dto